src/main/java/com/conveyal/analysis/components/HttpApi.java

```diff
@@ -79,14 +96,27 @@ public HttpApi (
             // Do not require authentication for internal API endpoints contacted by workers or for OPTIONS requests.
             String method = req.requestMethod();
             String pathInfo = req.pathInfo();
-            boolean authorize = pathInfo.startsWith("/api") && !"OPTIONS".equals(method);
+            boolean authorize = pathInfo.startsWith("/api") && !"OPTIONS".equalsIgnoreCase(method);
             if (authorize) {
-                // Determine which user is sending the request, and which permissions that user has. This method throws
-                // an error if the user cannot be authenticated.
-                UserPermissions user = authentication.authenticate(req);
-                LOG.info("{} {} by user {} in groups {}", method, pathInfo, user.email, user.groups);
+                // Determine which user is sending the request, and which permissions that user has.
+                // This method throws an exception if the user cannot be authenticated.
+                // Store the resulting permissions object in the request so it can be examined by any handler.
+                UserPermissions userPermissions = authentication.authenticate(req);
+                req.attribute(USER_PERMISSIONS_ATTRIBUTE, userPermissions);
+                // TODO stop using these two separate attributes, and use the permissions object directly
+                req.attribute(USER_EMAIL_ATTRIBUTE, userPermissions.email);
+                req.attribute(USER_GROUP_ATTRIBUTE, userPermissions.groups.iterator().next());
             }
-            BackendMain.recordActivityToPreventShutdown();
+        });
+
+        sparkService.after((req, res) -> {
+            // Firing an event after the request allows us to report the response time,
+            // but may fail to record requests experiencing authentication problems.
```

Comment:
Does `after` get skipped when an exception is thrown? 
Should we add a separate event to the `exception` handler that records the request time and the exception?

Reply:
I actually don't know, will have to dig in and see. I definitely intended to fire Exception events so we can show them on Slack and/or log them in the database. Will also ticket this one so we can follow up on it.